heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -5,39 +5,38 @@
 import './jestWorkerPatch';
 
 import * as path from 'path';
+import { getVersion, runCLI } from '@jest/core';
+import type { Config } from '@jest/types';
 import { resolveRunner, resolveSequencer, resolveTestEnvironment, resolveWatchPlugin } from 'jest-resolve';
 import { mergeWith, isObject } from 'lodash';
 import type {
-  ICleanStageContext,
-  IBuildStageContext,
-  IBuildStageProperties,
-  IPostBuildSubstage,
-  ITestStageContext,
-  ITestStageProperties,
-  IHeftPlugin,
   HeftConfiguration,
-  HeftSession,
-  ScopedLogger,
-  IHeftStringParameter,
-  IHeftFlagParameter,
-  IHeftIntegerParameter,
-  IHeftStringListParameter
+  IHeftTaskPlugin,
+  IHeftTaskSession,
+  IHeftTaskRunHookOptions,
+  IHeftTaskCleanHookOptions,
+  CommandLineFlagParameter,
+  CommandLineStringParameter
 } from '@rushstack/heft';
-import { getVersion, runCLI } from '@jest/core';
-import type { Config } from '@jest/types';
 import {
   ConfigurationFile,
-  IJsonPathMetadata,
+  type IJsonPathMetadata,
   InheritanceType,
   PathResolutionMethod
 } from '@rushstack/heft-config-file';
+import {
+  type ITypeScriptConfigurationJson,
+  type IPartialTsconfig,
+  loadTypeScriptConfigurationFileAsync,
+  loadPartialTsconfigFileAsync
+} from '@rushstack/heft-typescript-plugin';
```

Comment:
This makes me a little uncomfortable. Should this be an optional peer dependency? Seems like it'd be easy to end up in a situation with multiple versions of `heft-typescript-plugin`.